### PR TITLE
niri: add updateScript

### PR DESCRIPTION
**Pull request by Pandapip1**
## Description of changes

Adds a custom `updateScript` to keep `niri` up-to-date.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by IogaMaster — APPROVED** [review]

File: pkgs/by-name/ni/niri/package.nix [hunk 1]
```diff
@@ -1,6 +1,11 @@
 { lib
 , rustPlatform
 , fetchFromGitHub
+, writeShellApplication
+, common-updater-scripts
+, coreutils
+, git
+, nix
```
Comment (wegank) [inline pkgs/by-name/ni/niri/package.nix:8]:
```suggestion
, nix-update-script
```

File: pkgs/by-name/ni/niri/package.nix [hunk 2]
```diff
@@ -76,6 +81,32 @@ rustPlatform.buildRustPackage rec {
...
+      new_version="$(git tag --sort=-version:refname | head -n1)"
+      git checkout "$new_version"
+      new_version="$'' + ''{new_version#v}"
+      popd
+      update-source-version \
+          "$UPDATE_NIX_ATTR_PATH" \
+          "$new_version"
+      cp "$tmpdir/Cargo.lock" ./pkgs/by-name/ni/niri/Cargo.lock
+    '';
+  });
```
[the remark this answers is not in the document]
Comment (Pandapip1) [reply pkgs/by-name/ni/niri/package.nix:108]: `nix-update-script` does not update `Cargo.lock` files. A copy of `Cargo.lock` is necessary as there is a git-versioned dependency.

[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/by-name/ni/niri/package.nix:108]: There is a cargo updater called what? I forgot

[the remark this answers is not in the document]
Comment (wegank) [reply pkgs/by-name/ni/niri/package.nix:108]: > `nix-update-script` does not update `Cargo.lock` files. A copy of `Cargo.lock` is necessary as there is a git-versioned dependency.

It normally does. Try setting a lower version and replacing random bits in the hash and Cargo.lock to see if the update script fixes them.

[the remark this answers is not in the document]
Comment (Pandapip1) [reply pkgs/by-name/ni/niri/package.nix:108]: > It normally does. Try setting a lower version and replacing random bits in the hash and Cargo.lock to see if the update script fixes them.

If it does, that's undocumented behaviour.

[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/by-name/ni/niri/package.nix:108]: wegank is right, they work:

https://github.com/NixOS/nixpkgs/commit/10318b71e8132fd91b9341e6bcc06184897d5608

[the remark this answers is not in the document]
Comment (wegank) [reply pkgs/by-name/ni/niri/package.nix:108]: I guess it's worth creating an issue/PR at https://github.com/Mic92/nix-update for them to clarify the behavior of updating `cargoDeps` in readme.

File: pkgs/by-name/ni/niri/package.nix [hunk 3]
```diff
@@ -25,7 +26,7 @@ rustPlatform.buildRustPackage rec {
     owner = "YaLTeR";
     repo = "niri";
     rev = "v${version}";
-    hash = "sha256-EdlOGL9RdO77HnZxv2UpPwrJdFH8qPrPSRtkBBV167s=";
+    hash = "sha256-AzH1rZFqEH8sovZZfJykvsEmCedEZWigQFHWHl6/PdE=";
```
Comment (wegank) [inline pkgs/by-name/ni/niri/package.nix:29]:
```suggestion
    hash = "sha256-EdlOGL9RdO77HnZxv2UpPwrJdFH8qPrPSRtkBBV167s=";
```

**Pull request merged**